### DISCUSSION TITLE
Failure running the user provider startup check: error reading the authentication database: could not validate the schema: Users: non zero value required

### DISCUSSION THREAD

**Discussion by GVA-Guillaume** [authelia/authelia#5757] [Q&A] [answered]
### Version

v4.37.5

### Deployment Method

Docker

### Reverse Proxy

Caddy

### Description

Failure running the user provider startup check: error reading the authentication database: could not validate the schema: Users: non zero value required

aut_iam  | time="2023-08-02T20:41:49+02:00" level=info msg="Authelia v4.37.5 is starting"
aut_iam  | time="2023-08-02T20:41:49+02:00" level=error msg="Failure running the user provider startup check: error

### Reproduction

Deployment 

with  config file : 

storage: 
  encryption_key: a base 64 key
postgres:
    host:
   port: XXXXXX
   database: database
   schema: public
   username: root/user
  password: password

à chaque tentative, le schema de la base est bien créé et pourtant, impossible de poursuivre, donc la connexion fonctionne et les droits sont les bons

### Expectations

aut_iam  | time="2023-08-02T20:41:49+02:00" level=info msg="Authelia v4.37.5 is starting"
aut_iam  | time="2023-08-02T20:41:49+02:00" level=error msg="Failure running the user provider startup check: error reading the authentication database: could not validate the schema: Users: non zero value required" stack="github.com/authelia/authelia/v4/internal/commands/root.go:287 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  cmdRootRun\ngithub.com/spf13/cobra@v1.6.1/command.go:920                  (*Command).execute\ngithub.com/spf13/cobra@v1.6.1/command.go:1044                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.1/command.go:968                  (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/proc.go:250                                           main\nruntime/asm_amd64.s:1594                                      goexit"
aut_iam  | time="2023-08-02T20:41:50+02:00" level=fatal msg="The following providers had fatal failures during startup: user" stack="github.com/authelia/authelia/v4/internal/commands/root.go:309 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  cmdRootRun\ngithub.com/spf13/cobra@v1.6.1/command.go:920                  (*Command).execute\ngithub.com/spf13/cobra@v1.6.1/command.go:1044                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.1/command.go:968                  (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/proc.go:250                                           main\nruntime/asm_amd64.s:1594                                      goexit"

### Configuration (Authelia)

```yaml
storage: 
  encryption_key: a base 64 key
postgres:
    host:
   port: XXXXXX
   database: database
   schema: public
   username: root/user
  password: password

à chaque tentative, le schema de la base est bien créé et pourtant, impossible de poursuivre, donc la connexion fonctionne et les droits sont les bons
```

### Logs (Authelia)

```shell
aut_iam  | time="2023-08-02T20:41:49+02:00" level=info msg="Authelia v4.37.5 is starting"
aut_iam  | time="2023-08-02T20:41:49+02:00" level=error msg="Failure running the user provider startup check: error reading the authentication database: could not validate the schema: Users: non zero value required" stack="github.com/authelia/authelia/v4/internal/commands/root.go:287 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  cmdRootRun\ngithub.com/spf13/cobra@v1.6.1/command.go:920                  (*Command).execute\ngithub.com/spf13/cobra@v1.6.1/command.go:1044                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.1/command.go:968                  (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/proc.go:250                                           main\nruntime/asm_amd64.s:1594                                      goexit"
aut_iam  | time="2023-08-02T20:41:50+02:00" level=fatal msg="The following providers had fatal failures during startup: user" stack="github.com/authelia/authelia/v4/internal/commands/root.go:309 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:87  cmdRootRun\ngithub.com/spf13/cobra@v1.6.1/command.go:920                  (*Command).execute\ngithub.com/spf13/cobra@v1.6.1/command.go:1044                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.6.1/command.go:968                  (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\nruntime/proc.go:250                                           main\nruntime/asm_amd64.s:1594                                      goexit"
```

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] This is a bug report and not a support request

- [X] I have read the security policy and this bug report is not a security issue or security related issue

- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration

- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide

- [X] I have checked for related proxy or application logs and included them if available

- [X] I have checked for related issues and checked the documentation

**Comment by james-d-elliott** [comment, marked as answer]
1. You need to configure a user in the authentication backend YAML file.
2. Your configuration is invalid:
    1. There are several indentation issues.
    2. You're missing 90% of the required configuration.
3. The logs (debug/trace) are incomplete I think.
4. No clear bug has been demonstrated.

For these reasons I'm converting this to a Q/A as it's a user support request, Not a bug.

**Reply by GVA-Guillaume** [reply]
Hi, what do you mean to create a user ? in the yaml file except host, database, username, password ? do you have a ref. link about what do you mean ?

Are you saying I need to authenticate my database connection from my ldap directory ? this is what you mean ?

the indentation issues are the fact about copy past without values
the rest of the configuration perfectly, authelia is up, if I don't use storage configuration (see sqlite)

i don't have more trace about incindet (I precise I am on trace level)

**Reply by james-d-elliott** [reply]
Sorry for not responding sooner. See the following guides (you have configured the file auth backend according to the logs, but it's impossible tell what's exactly going on as the full configuration which was requested wasn't provided):

- https://www.authelia.com/configuration/first-factor/file/
- https://www.authelia.com/reference/guides/passwords/#yaml-format
